src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -129,7 +129,7 @@ func (mod *mgmtModule) handleGetAttachInfo(ctx context.Context, reqb []byte, pid
 	}
 
 	var err error
-	numaNode := mod.aiCache.defaultNumaNode
+	var numaNode int
 
 	if mod.numaAware {
 		numaNode, err = netdetect.GetNUMASocketIDForPid(mod.netCtx, pid)
```

Comment:
There is still an unsynchronized write here. I think you'll need to move the lock up to protect it.